projects/go-lib/src/lib/components/go-file-upload/go-file-upload.component.ts

```diff
@@ -59,6 +62,25 @@ export class GoFileUploadComponent implements OnInit {
     }
   }
 
+  onDragOver(): void {
+    if (!this.isLoading) {
+      this.state = 'selecting';
+    }
+  }
+
+  onDragLeave(): void {
+    if (this.files.length) {
+      this.state = 'selected';
+    }
+  }
+
+  reset(): void {
+    this.files.reset();
```

Comment:
Yeah, I didn't really think that all the way through I guess. I can remove the `removeAt(0)` in favor of using `this.files.clear()` in the `reset()` method. This will clear all `control`s in the `formArray` where `this.files.reset()` make all the `control`s values `null`.